### ISSUE TITLE
requiring "bundler/setup" can modify Gemfile.lock

### ISSUE THREAD

**Issue by wconrad** [bundler/bundler#2710]
Requiring "bundler/setup" can modify _Gemfile.lock_.  In a separate communication with @mpapis, we learned that it should not.

[This gist](https://gist.github.com/wconrad/7435134) contains a script to reproduce the problem, and other relevant files.  Running that script shows (on my system) that _Gemfile.lock_ can be modified by `require "bundler/setup"`.

**Comment by wconrad** [comment]
A quick note about my issue of losing gems from Gemfile.lock.  The issue still occurs for my company, so I went back to do some more digging.  I am able to reproduce it on a box running Debian "wheezy" (the previous stable version), but not on one running Debian "jessie" (the current stable version).  This is with identical commands, same version of Ruby, latest version of Bundler, etc.  It seems likely that Bundler was never at fault (we never doubted ya, Bundler), but that the cause was a bug in the kernel or in some C library that was fixed at some point.

Just in case you were worried about Bundler at all.  And in case anyone else with a similar issue runs across this.
